pl_examples/models/allennlp_template.py

```diff
@@ -0,0 +1,214 @@
+        output_dict = self.model.make_output_human_readable(output_dict)
+
+        if 'loss' not in output_dict:
+            raise ValueError("AllenNlp model returned dictionary missing the 'loss' key")
+
+        # AllenNlp models are responsible for their own metrics
+        output_dict.update(self.model.get_metrics())
+
+        update = {}
+        update['log'] = {k: v for k, v in output_dict.items() if is_loggable(v)}
+        update['log']['loss'] = output_dict['loss']
+        update['log']['epoch'] = self.current_epoch
+        #         update['log']['batch'] = batch_idx
+        update['progress_bar'] = {k: v for k, v in output_dict.items()
+                                  if is_scalar(v) and k != 'loss'}
+        update['progress_bar']['batch'] = batch_idx
+        output_dict.update(update)
+        return output_dict
+
+    def training_step(self, batch, batch_idx):
+        return self._step(batch, batch_idx)
+
+    def validation_step(self, batch, batch_idx):
+        return self._step(batch, batch_idx)
+
+    def validation_epoch_end(self, outputs):
+        result = {'log': {}, 'progress_bar': {}}
+
+        # Handle loss:
+        val_loss = torch.stack([x['loss'] for x in outputs]).mean()
+        result = {'log': {}, 'progress_bar': {}}
+        result['val_loss'] = val_loss
+        result['log']['val_loss'] = val_loss
+        result['progress_bar']['val_loss'] = val_loss
+
+        # Handle everything else
+        for key in outputs[0].keys():
+            if key in ['log', 'progress_bar']:
+                continue
+            val_key = "val_" + key
+            if is_scalar(outputs[0][key]):
+                # take mean of scalar values
+                if isinstance(outputs[0][key], torch.Tensor):
+                    val = torch.stack([x[key] for x in outputs]).mean()
+                else:
+                    val = statistics.mean([x[key] for x in outputs])
+                result[val_key] = val
+                result['log'][val_key] = val
+                result['progress_bar'][val_key] = val
+            elif is_loggable(outputs[0][key]):
+                # chain non-scalar values into a list
+                val = list(itertools.chain(*[output[key] for output in outputs]))
+                result['log'][val_key] = val
+
+        # Add tracking variables
+        result['log']['epoch'] = self.current_epoch
+        return result
+
+    @pytorch_lightning.data_loader
+    def train_dataloader(self):
+        return self.data_loader
+
+    @pytorch_lightning.data_loader
+    def val_dataloader(self):
+        return self.validation_data_loader
+
+    # Not used for now, since we initialize training parameters using Pytorch-Lightning
+    def configure_optimizers(self):
+        from allennlp.training.optimizers import Optimizer
+        parameters = [[n, p] for n, p in self.model.named_parameters() if p.requires_grad]
+        return Optimizer.from_params(
+            params=params['trainer']['optimizer'],
```

Comment:
`params` is not defined

Reply:
I must have checked in an incorrect version. Please change line 210 to `params = ...` I no longer have access to my setup for coding